src/dapp/walletProvider.ts

```diff
@@ -359,6 +164,47 @@ export class WalletProvider implements IDappProvider {
         throw new ErrNotImplemented();
     }
 
+    static isTxSignReturnSuccess(urlParams: any): boolean {
+        return urlParams.hasOwnProperty(WALLET_PROVIDER_CALLBACK_PARAM) && urlParams[WALLET_PROVIDER_CALLBACK_PARAM] === WALLET_PROVIDER_CALLBACK_PARAM_TX_SIGNED;
+    }
+
+    static getTxSignReturnValue(urlParams: any): Transaction[] {
+        const expectedProps = ["nonce", "value", "receiver", "sender", "gasPrice",
+            "gasLimit", "data", "chainID", "version", "signature"];
```

Comment:
`options` ?

Reply:
The actual wallet does not take `options` into consideration and never sends it back. This would be another task, starting with the wallet hooks, then adapting `erdjs`